File: Cpp/Problem_1-10/Problem_1B/Program.cpp
#include <iostream>
#include <regex>
using namespace std;

int main()
{
    int n;
    cin >> n;
    for (int i = 0; i < n; i++)
    {
        string cell;
        cin >> cell;
        if (cell[0] == 'R' && isdigit(cell[1]) && cell.find('C') != string::npos)
        {
            int col = 0, row = 0;
            int cPos = cell.find('C');
            row = stoi(cell.substr(1, cPos - 1));
            col = stoi(cell.substr(cPos + 1));

            string column;
            while (col > 0)
            {
                col--;
                column += 'A' + (col % 26);
                col /= 26;
            }
            reverse(column.begin(), column.end());
            cout << column << row << '\n';
        }
        else
        {
            int col = 0, row = 0, j = 0;
            while (isalpha(cell[j]))
            {
                col = col * 26 + (cell[j] - 'A' + 1);
                j++;
            }
            row = stoi(cell.substr(j));

            cout << "R" << row << "C" << col << '\n';
        }
    }
}

File: CSharp/Problem_1-10/Problem_1B/Program.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Problem_1B
{
    internal class Program
    {
        static void Main(string[] args)
        {
            int t = int.Parse(Console.ReadLine());

            for (int i = 0; i < t; i++)
            {
                string cell = Console.ReadLine().Trim();
                if (cell[0] == 'R' && char.IsDigit(cell[1]) && cell.Contains("C"))
                {
                    // Định dạng: RxCy
                    int cIndex = cell.IndexOf('C');
                    string row = cell.Substring(1, cIndex - 1);
                    int col = int.Parse(cell.Substring(cIndex + 1));

                    // Chuyển đổi số cột thành chữ cái
                    List<char> column = new List<char>();
                    while (col > 0)
                    {
                        col--; // Giảm cột đi 1 để chuyển sang chỉ số 0
                        column.Add((char)(col % 26 + 'A'));
                        col /= 26;
                    }

                    column.Reverse();
                    Console.WriteLine(string.Join("", column) + row);
                }
                else
                {
                    // Định dạng: AxY
                    int col = 0;
                    int j = 0;

                    while (j < cell.Length && char.IsLetter(cell[j]))
                    {
                        col = col * 26 + (cell[j] - 'A' + 1);
                        j++;
                    }

                    string row = cell.Substring(j);
                    Console.WriteLine($"R{row}C{col}");
                }
            }
        }
    }
}